check.py

```diff
@@ -314,127 +295,126 @@ def run_spec_tests():
         # FIXME we support old and new memory formats, for now, until 0xc, and so can't pass this old-style test.
         BLACKLIST = ['binary.wast']
         # FIXME to update the spec to 0xd, we need to implement (register "name") for import.wast
-        spec_tests = [os.path.join('spec', t) for t in sorted(os.listdir(os.path.join(options.binaryen_test, 'spec'))) if t not in BLACKLIST]
+        spec_tests = get_tests(SPEC_TEST_DIR, ['wast'])
+        spec_tests = [t for t in spec_tests if os.path.basename(t) not in BLACKLIST]
     else:
         spec_tests = options.spec_tests[:]
 
-    for t in spec_tests:
-        if t.startswith('spec') and t.endswith('.wast'):
-            print('..', t)
-            wast = os.path.join(options.binaryen_test, t)
+    for wast in spec_tests:
+        print('..', os.path.basename(wast))
 
-            # skip checks for some tests
-            if os.path.basename(wast) in ['linking.wast', 'nop.wast', 'stack.wast', 'typecheck.wast', 'unwind.wast']:    # FIXME
-                continue
+        # skip checks for some tests
+        if os.path.basename(wast) in ['linking.wast', 'nop.wast', 'stack.wast', 'typecheck.wast', 'unwind.wast']:    # FIXME
+            continue
+
+        def run_spec_test(wast):
+            cmd = WASM_SHELL + [wast]
+            # we must skip the stack machine portions of spec tests or apply other extra args
+            extra = {
+            }
```

Comment:
This can be on a single line (although I see you didn't change that land really).

Reply:
Done